src/DebugEngineHost/HostConfigurationStore.cs

```diff
@@ -88,14 +88,13 @@ public void GetExceptionCategorySettings(Guid categoryId, out HostConfigurationS
         /// </summary>
         /// <param name="enableLoggingSettingName">[Optional] In VS, the name of the settings key to check if logging is enabled. If not specified, this will check 'Logging' in the AD7 Metrics.</param>
         /// <param name="logFileName">[Required] name of the log file to open if logging is enabled.</param>
-        /// <returns>[Optional] If logging is enabled, the logging object.</returns>
+        /// <returns>If no error then logging object. If file cannot be openened then throw an exception. Otherwise return an empty logger - the user can explictly reconfigure it later</returns>
         public HostLogger GetLogger(string enableLoggingSettingName, string logFileName)
         {
             if (string.IsNullOrEmpty(logFileName))
             {
-                throw new ArgumentNullException("logFileName");
+                return new HostLogger();    // empty logger
```

Comment:
If we are going to switch loggers then additional locking is necessary in the Logger to prevent message losses on switching logs. Or maybe that just isn't an issue we care about?